Author: Alastair McKinstry <mckinstry@debian.org>
Description: make check should die with exit 1 on a test failure, so that
 Debian's automated testing notices.
Last-Updated: 2012-06-30
Forwarded: no

Index: slang2-2.2.4/src/test/Makefile
===================================================================
--- slang2-2.2.4.orig/src/test/Makefile	2012-06-30 03:08:03.000000000 +0100
+++ slang2-2.2.4/src/test/Makefile	2012-06-30 03:09:35.000000000 +0100
@@ -19,10 +19,10 @@
 	@echo ""
 	@echo "Running tests:"
 	@echo ""
-	-@for X in $(TEST_SCRIPTS); \
+	@for X in $(TEST_SCRIPTS); \
 	do \
-	    $(RUN_TEST_PGM) $$X.sl; \
-	    $(RUN_TEST_PGM) -utf8 $$X.sl; \
+	    if ! eval $(RUN_TEST_PGM) $$X.sl; then exit 1; fi ;  \
+	    if ! eval $(RUN_TEST_PGM) -utf8 $$X.sl; then exit 1 ; fi  \
 	done
 	touch sltest.c
 
@@ -30,10 +30,10 @@
 	@echo ""
 	@echo "Running tests:"
 	@echo ""
-	-@for X in $(TEST_SCRIPTS_SLC); \
+	@for X in $(TEST_SCRIPTS_SLC); \
 	do \
-	    $(RUN_TEST_PGM) $$X.slc; \
-	    $(RUN_TEST_PGM) -utf8 $$X.slc; \
+	    if ! eval $(RUN_TEST_PGM) $$X.slc; then exit 1; fi ;  \
+	    if ! eval $(RUN_TEST_PGM) -utf8 $$X.slc; then exit 1; fi ; \
 	done
 	touch sltest.c
 
